src/Symfony/Bundle/FrameworkBundle/Logger/Logger.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Logger;
+
+use Psr\Log\AbstractLogger;
+use Psr\Log\InvalidArgumentException;
+use Psr\Log\LogLevel;
+
+/**
+ * Minimalist PSR-3 logger designed to write in stdout, stderr or any other stream.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+final class Logger extends AbstractLogger
+{
+    private static $defaultOutputByLevel = array(
+        LogLevel::EMERGENCY => 'php://stderr',
+        LogLevel::ALERT => 'php://stderr',
+        LogLevel::CRITICAL => 'php://stderr',
+        LogLevel::ERROR => 'php://stderr',
+        LogLevel::WARNING => 'php://stderr',
+        LogLevel::NOTICE => false,
+        LogLevel::INFO => false,
+        LogLevel::DEBUG => false,
+    );
+    private $outputByLevel;
+    private $format;
+    private $dateFormat;
+    private $handles = array();
+
+    public function __construct(array $outputByLevel = array(), $format = '[%s] %s', $dateFormat = \DateTime::RFC3339)
+    {
+        $this->outputByLevel = $outputByLevel + self::$defaultOutputByLevel;
+        $this->format = $format;
+        $this->dateFormat = $dateFormat;
+    }
+
+    public function __destruct()
+    {
+        foreach ($this->handles as $handle) {
+            fclose($handle);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function log($level, $message, array $context = array())
+    {
+        if (!isset($this->outputByLevel[$level])) {
+            throw new InvalidArgumentException(sprintf('The log level "%s" does not exist.', $level));
+        }
+
+        if ($stream = $this->outputByLevel[$level]) {
+            if (!isset($this->handles[$stream])) {
+                $this->handles[$stream] = fopen($stream, 'a');
+            }
+
+            fwrite($this->handles[$stream], sprintf($this->format, $level, $this->interpolate($message, $context)).PHP_EOL);
+        }
+    }
+
+    /**
+     * Interpolates context values into the message placeholders.
+     *
+     * @author Jordi Boggiano <j.boggiano@seld.be>
+     *
+     * @see https://github.com/Seldaek/monolog/blob/master/src/Monolog/Processor/PsrLogMessageProcessor.php
+     *
+     * @param string $message
+     *
+     * @return string
+     */
+    private function interpolate($message, array $context)
```

Comment:
This PR is awesome ! But any chance to move this behavior in a dedicated class behind interface ? 
Like the Formatter of Monolog. I know you want to keep it simple but it is a big lack to be used imo.

We look for this kind of logger as it is our main usage (docker). But our logs are parsed into an ELK and without custom formatting we will not be able to use it (or change all other apps not in PHP that output logs and logstash config...).

Reply:
Maybe can we just allow a callable in `format` instead of a string?

Reply:
`interpolate` is added here only because it is part of the PSR-3 standard AFAIK. For custom advanced formatting, Monolog should be used

Reply:
I'm not sure a callable only for `format` would help (or I miss anything?). 

We would need both message and context separate as @ogizanagi said (I just realized he said the same thing).

I don't see any problem to not having this behavior embed by default, but if we could make it possible by design, we will definitively use it. Like a callable on the whole sprintf with `message` `level` `context` as arguments ?

But apparently other people are not enthusiastic about adding more configuration :p I guess we could just copy/paste the code also.

Reply:
The callable will take the message and the context in parameter, and return a string. if none is passed, the current `sprintf` will be used.

It has several benefits:
- on one hand it's very flexible for the end user, allowing all kind of customizations
- on the other hand, we can remove all configuration options and it's very solid

WDYT @fabpot @javiereguiluz?

Reply:
No configuration seems better to me. Any configuration should involve using Monolog.

Reply:
Agreed. Let's forgot about this.

Reply:
I've added a callable formatter implementation in a separate commit. It's just 1 line of code but it opens interesting possibility (see the json formatter in tests) in a cloud environment. IMO it is worth it.

Reply:
Without the formatter callable, I guess everybody will keep using monolog in fact, all the time. So in this case I'm not sure it's a good move to merge this PR at all.

Either monolog remains the log dependency - no need for this Logger - either the formatter should be added and we can sometimes decide to require monolog or not depending of the complexity of the wanted log stack. WDYT?